### PR TITLE
#19291 unifying the 3 quartz schedulers

### PR DESCRIPTION
**Pull request by wezell**
To see the tests.
This is meant to simplify quartz where we're removing repetitive schedulers there were adding up a pool of unnecessary threads. 
it is coupled with an enterprise PR https://github.com/dotCMS/enterprise-2.x/pull/753

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImplIntegrationTest.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 import com.dotmarketing.portlets.templates.design.bean.TemplateLayout;
 import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.quartz.DotSchedulerFactory;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImplIntegrationTest.java:21]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java [hunk 2]
```diff
@@ -502,11 +444,10 @@ else if (sched.getTrigger(triggerName, triggerGroup) != null)
...
 	 */
 	public static boolean removeJob(String jobName, String jobGroup) throws SchedulerException {
-		Scheduler sched = DotSchedulerFactory.getInstance().getSequentialScheduler();
+		Scheduler sched = DotSchedulerFactory.getInstance().getScheduler();
 		boolean result1 = removeJob(jobName, jobGroup, sched);
-		sched = DotSchedulerFactory.getInstance().getScheduler();
-		boolean result2 = removeJob(jobName, jobGroup, sched);
-		return result1 | result2;
+
+		return result1 ;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:450]: Codacy found an issue: [Consider simply returning the value vs storing it in local variable 'result1'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java [hunk 3]
```diff
@@ -679,11 +561,8 @@ private static void pauseTrigger(String triggerName, String triggerGroup, Schedu
 	 * @throws SchedulerException
 	 */
 	public static Trigger getTrigger(String triggerName, String triggerGroup) throws SchedulerException {
-		Trigger t = getSequentialScheduler ().getTrigger(triggerName, triggerGroup);
-		if(t==null){
-			t = getStandardScheduler () .getTrigger(triggerName, triggerGroup);
-			
-		}
+		Trigger t = t = DotSchedulerFactory.getInstance().getScheduler() .getTrigger(triggerName, triggerGroup);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:564]: Codacy found an issue: [Avoid variables with short names like t](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:450]: Codacy found an issue: [Consider simply returning the value vs storing it in local variable 'result1'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImplIntegrationTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImplIntegrationTest.java:21]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:564]: Codacy found an issue: [Avoid variables with short names like t](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+<%@page import="java.util.List"%>
+<%@page import="com.dotmarketing.business.APILocator"%>
+<%@ page import="com.dotmarketing.util.UtilMethods" %>
+<%@ page import="com.dotmarketing.beans.Host" %>
+<%@ page import="com.dotmarketing.business.web.WebAPILocator"%>
+
+<%@ include file="/html/common/uservalidation.jsp"%>
+
+<%String[] groups = QuartzUtils.getScheduler().getJobGroupNames(); %>
+<table class="listingTable">
```
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:24]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      					<%if(CronTrigger.MISFIRE_INSTRUCTION_DO_NOTHING == trig.getMisfireInstruction()){ %>
+      						<%= LanguageUtil.get(pageContext, "scheduler.job.misfire.donothing") %>
+      					<%}else if(CronTrigger.MISFIRE_INSTRUCTION_FIRE_ONCE_NOW == trig.getMisfireInstruction()){ %>
+      						<%= LanguageUtil.get(pageContext, "scheduler.job.misfire.fireOnce") %>
+      					<%}%>
+      				<%} %>
+      			</td>
+      		</tr>
+		<%}catch(Exception e){%>
+		<tr><td><%=t%></td><td colspan="10" class="red"><%=LanguageUtil.get(pageContext, "an-unexpected-error-occurred")+"<br/>"+e.getMessage() %></td>
```
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:80]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:450]: Codacy found an issue: [Consider simply returning the value vs storing it in local variable 'result1'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:80]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:24]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:564]: Codacy found an issue: [Avoid variables with short names like t](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImplIntegrationTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImplIntegrationTest.java:21]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java [hunk 6]
```diff
@@ -110,7 +110,7 @@ public void Test_Launch_Stateful_Jobs_Verify_They_Dont_Overlap_In_Time()
     private Optional<JobExecutionContext> getJobExecutionContext(){
         final String jobName = getJobName(MyStatefulJob.class);
         try {
-            final Scheduler sequentialScheduler = getSequentialScheduler();
+            final Scheduler sequentialScheduler = DotSchedulerFactory.getInstance().getScheduler();
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java:113]: Nitpick: Do we still need to name it `sequentialScheduler`?

Comment (wezell) [reply dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java:113]: Huum, well, I think the scheduler acts as Sequential if the job is a StatefulJob and just a scheduler if the job is a Job.

File: dotCMS/src/main/java/com/dotmarketing/init/DotInitScheduler.java [hunk 7]
```diff
@@ -755,7 +550,7 @@ private static void addSystemEventsJob () {
 	 */
 	private static void scheduleJob(final JobBuilder jobBuilder) {
 		try {
-			Scheduler sched = QuartzUtils.getStandardScheduler();
+			Scheduler sched = DotSchedulerFactory.getInstance().getScheduler();
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/init/DotInitScheduler.java:553]: Above there are references to `QuartzUtils.getScheduler()`. I wonder if they are the same and i f they are, we should pick one.

File: dotCMS/src/main/java/com/dotmarketing/quartz/DotSchedulerFactory.java [hunk 8]
```diff
@@ -5,107 +5,67 @@
...
+            }
+            standardSchedulerFactory = new StdSchedulerFactory(standardProperties);
+            standardScheduler = standardSchedulerFactory.getScheduler();
+        }
+
+
+    }
+
+    public Collection<Scheduler> getAllSchedulers() throws SchedulerException {
+        List<Scheduler> list = new ArrayList<Scheduler>();
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/DotSchedulerFactory.java:44]: Nitpick: return instead `Arrays.asList(...)`

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java [hunk 9]
```diff
@@ -775,79 +616,23 @@ public static void startSchedulers () throws SchedulerException {
...
-     * Starts the local quartz scheduler
-     *
-     * @throws SchedulerException
-     */
-    public static void startLocalScheduler () throws SchedulerException {
-        Scheduler localScheduler = DotSchedulerFactory.getInstance().getLocalScheduler();
-        localScheduler.start();
-    }
+	
+	public static boolean isJobRunning(String jobName, String jobGroup) throws SchedulerException{
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:626]: @wezell when this gets merge you will notice a new `isJobRunning()` method here: https://github.com/dotCMS/core/pull/19273/commits/704c8c6ebe7d5c3cc85438796c12b4b95b69461b#diff-834b3711e1888aa36bedbb65f457e411R870

File: dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java (hunk 6 above)
Comment (fabrizzio-dotCMS) [inline dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java:113]: Theoretically this class `CronScheduledTask` has a param that should instruct before what scheduler needed to be used. Sequential/Standart maybe that para  and the getter should be removed or at least marked deprecated

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java [hunk 10]
```diff
@@ -40,29 +37,7 @@
...
-	}
-	
-	/**
-	 * 
-	 * Lists all jobs scheduled through the standard scheduler, the standard scheduler
-	 * let you run multiple jobs in parallel
-	 * 
-	 * @return
-	 */
-	public static List<ScheduledTask> getStandardScheduledTasks() throws SchedulerException {
```
Comment (fabrizzio-dotCMS) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:62]: I get a `java.lang.NoSuchMethodError: com.dotmarketing.quartz.QuartzUtils.getStandardScheduledTasks` from site search on this branch

**Review by fabrizzio-dotCMS — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/servlets/RunScheduledJobServlet.java [hunk 11]
```diff
@@ -18,6 +18,7 @@
 import com.dotmarketing.business.Role;
 import com.dotmarketing.business.RoleAPI;
 import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.quartz.DotSchedulerFactory;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/servlets/RunScheduledJobServlet.java:21]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:24]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherAPIImpl.java [hunk 12]
```diff
@@ -23,6 +23,7 @@
 import com.dotmarketing.portlets.folders.model.Folder;
 import com.dotmarketing.portlets.structure.factories.StructureFactory;
 import com.dotmarketing.portlets.structure.model.Structure;
+import com.dotmarketing.quartz.DotSchedulerFactory;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/business/PublisherAPIImpl.java:26]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java [hunk 13]
```diff
@@ -430,35 +439,31 @@ public static void removeTaskRuntimeValues(String jobName, String jobGroup) {
...
-		Date endDate = job.getEndDate();
+		final String jobName = job.getJobName();
+		final String jobGroup = job.getJobGroup();
+		final String triggerName = job.getTriggerName() == null ? jobName + "_trigger" : job.getTriggerName();
+		final String triggerGroup = job.getTriggerGroup() == null ? jobGroup : job.getTriggerGroup();
+		final Date startDate = job.getStartDate();
+		final Date endDate = job.getEndDate();
 
-		if ((jobDetail = sched.getJobDetail(jobName, jobGroup)) == null) {
+		if ((jobDetail = scheduler.getJobDetail(jobName, jobGroup)) == null) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:459]: Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java [hunk 14]
```diff
@@ -472,17 +477,17 @@ public static void scheduleTask(ScheduledTask job) throws SchedulerException, Pa
...
-		sched.addJob(jobDetail, true);
+		scheduler.addJob(jobDetail, true);
 
 		if (isNew)
-			sched.scheduleJob(trigger);
-		else if (sched.getTrigger(triggerName, triggerGroup) != null)
-			sched.rescheduleJob(triggerName, triggerGroup, trigger);
+			scheduler.scheduleJob(trigger);
+		else if (scheduler.getTrigger(triggerName, triggerGroup) != null)
+			scheduler.rescheduleJob(triggerName, triggerGroup, trigger);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:485]: Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java [hunk 15]
```diff
@@ -472,17 +477,17 @@ public static void scheduleTask(ScheduledTask job) throws SchedulerException, Pa
...
 		trigger.setMisfireInstruction(job.getMisfireInstruction());
 
-		sched.addJob(jobDetail, true);
+		scheduler.addJob(jobDetail, true);
 
 		if (isNew)
-			sched.scheduleJob(trigger);
-		else if (sched.getTrigger(triggerName, triggerGroup) != null)
-			sched.rescheduleJob(triggerName, triggerGroup, trigger);
+			scheduler.scheduleJob(trigger);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:483]: Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java [hunk 16]
```diff
@@ -501,12 +506,11 @@ else if (sched.getTrigger(triggerName, triggerGroup) != null)
...
-		Scheduler sched = DotSchedulerFactory.getInstance().getSequentialScheduler();
-		boolean result1 = removeJob(jobName, jobGroup, sched);
-		sched = DotSchedulerFactory.getInstance().getScheduler();
-		boolean result2 = removeJob(jobName, jobGroup, sched);
-		return result1 | result2;
+	public static boolean removeJob(final String jobName, final String jobGroup) throws SchedulerException {
+		final Scheduler scheduler = getScheduler();
+		boolean result1 = removeJob(jobName, jobGroup, scheduler);
+
+		return result1 ;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:513]: Codacy found an issue: [Consider simply returning the value vs storing it in local variable 'result1'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotcms/publisher/pusher/PushPublisher.java [hunk 17]
```diff
@@ -49,6 +49,7 @@
 import com.dotcms.util.CloseUtils;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.cms.factories.PublicEncryptionFactory;
+import com.dotmarketing.quartz.DotSchedulerFactory;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/pusher/PushPublisher.java:52]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImplIntegrationTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImplIntegrationTest.java:21]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/job/IdentifierDateJob.java [hunk 18]
```diff
@@ -33,6 +33,7 @@
 import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
 import com.dotmarketing.portlets.contentlet.model.ContentletVersionInfo;
 import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.quartz.DotSchedulerFactory;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/job/IdentifierDateJob.java:36]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:80]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java (hunk 14 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:485]: Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:80]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java (hunk 13 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:459]: Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/servlets/RunScheduledJobServlet.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/servlets/RunScheduledJobServlet.java:21]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/portlets/scheduler/action/ViewSchedulersAction.java [hunk 19]
```diff
@@ -29,6 +29,7 @@
 import com.dotcms.repackage.org.apache.struts.action.ActionForm;
 import com.dotcms.repackage.org.apache.struts.action.ActionForward;
 import com.dotcms.repackage.org.apache.struts.action.ActionMapping;
+import com.dotmarketing.quartz.DotSchedulerFactory;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/scheduler/action/ViewSchedulersAction.java:32]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotcms/publisher/pusher/PushPublisher.java (hunk 17 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/pusher/PushPublisher.java:52]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherAPIImpl.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/business/PublisherAPIImpl.java:26]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/job/IdentifierDateJob.java (hunk 18 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/job/IdentifierDateJob.java:36]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java (hunk 15 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:483]: Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:24]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java (hunk 16 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:513]: Codacy found an issue: [Consider simply returning the value vs storing it in local variable 'result1'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/portlets/scheduler/action/ViewSchedulersAction.java (hunk 19 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/scheduler/action/ViewSchedulersAction.java:32]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:24]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/job/IdentifierDateJob.java (hunk 18 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/job/IdentifierDateJob.java:36]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherAPIImpl.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/business/PublisherAPIImpl.java:26]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/servlets/RunScheduledJobServlet.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/servlets/RunScheduledJobServlet.java:21]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java (hunk 13 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:459]: Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:80]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotcms/publisher/pusher/PushPublisher.java (hunk 17 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/pusher/PushPublisher.java:52]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.quartz.DotSchedulerFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

**Review by fabrizzio-dotCMS — APPROVED** [review]

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:24]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:80]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java [hunk 20]
```diff
@@ -427,35 +439,31 @@ public static void removeTaskRuntimeValues(String jobName, String jobGroup) {
...
-		Date endDate = job.getEndDate();
+		final String jobName = job.getJobName();
+		final String jobGroup = job.getJobGroup();
+		final String triggerName = job.getTriggerName() == null ? jobName + "_trigger" : job.getTriggerName();
+		final String triggerGroup = job.getTriggerGroup() == null ? jobGroup : job.getTriggerGroup();
+		final Date startDate = job.getStartDate();
+		final Date endDate = job.getEndDate();
 
-		if ((jobDetail = sched.getJobDetail(jobName, jobGroup)) == null) {
+		if ((jobDetail = scheduler.getJobDetail(jobName, jobGroup)) == null) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:459]: Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:24]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsmaintenance/system_jobs.jsp:80]: Codacy found an issue: [Do not use an attribute called class.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

File: dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java (hunk 20 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/quartz/QuartzUtils.java:459]: Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6213816)

**Pull request merged**